packages/trpc/server/routers/viewer/organizations/create.handler.ts

```diff
@@ -0,0 +1,132 @@
+import { createHash } from "crypto";
+import { totp } from "otplib";
+
+import { sendOrganizationEmailVerification } from "@calcom/emails";
+import { hashPassword } from "@calcom/features/auth/lib/hashPassword";
+import { subdomainSuffix } from "@calcom/features/ee/organizations/lib/orgDomains";
+import { IS_PRODUCTION, IS_TEAM_BILLING_ENABLED } from "@calcom/lib/constants";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import { prisma } from "@calcom/prisma";
+import { MembershipRole } from "@calcom/prisma/enums";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TrpcSessionUser } from "../../../trpc";
+import type { TCreateInputSchema } from "./create.schema";
+
+type CreateOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TCreateInputSchema;
+};
+
+const vercelCreateDomain = async (domain: string) => {
+  const response = await fetch(
+    `https://api.vercel.com/v8/projects/${process.env.PROJECT_ID_VERCEL}/domains?teamId=${process.env.TEAM_ID_VERCEL}`,
+    {
+      body: `{\n  "name": "${domain}.${subdomainSuffix()}"\n}`,
+      headers: {
+        Authorization: `Bearer ${process.env.AUTH_BEARER_TOKEN_VERCEL}`,
+        "Content-Type": "application/json",
+      },
+      method: "POST",
+    }
+  );
+
+  const data = await response.json();
+
+  // Domain is already owned by another team but you can request delegation to access it
+  if (data.error?.code === "forbidden")
+    throw new TRPCError({ code: "CONFLICT", message: "domain_taken_team" });
+
+  // Domain is already being used by a different project
+  if (data.error?.code === "domain_taken")
+    throw new TRPCError({ code: "CONFLICT", message: "domain_taken_project" });
+
+  return true;
+};
+
+export const createHandler = async ({ input }: CreateOptions) => {
+  const { slug, name, adminEmail, adminUsername, check } = input;
+
+  const userCollisions = await prisma.user.findFirst({
+    where: {
+      email: adminEmail,
+    },
+  });
+
+  const slugCollisions = await prisma.team.findFirst({
```

Comment:
Can anyone reserve an organization slug? What happens if someone not from acme(doesn't even have acme verified email), uses acme slug. Do we want to prevent it?

Reply:
Anyone can as this will be part of the Enterprise tier and handled by us mainly first, not self-served. But even if it was public, we don't mind anyone squatting org slugs, probably not going to happen either way.